src/hyperlight_host/src/hypervisor/mod.rs

```diff
@@ -536,14 +734,28 @@ impl LinuxInterruptHandle {
         (running, generation)
     }
 
-    fn send_signal(&self) -> bool {
+    fn send_signal(&self, stamp_generation: bool) -> bool {
         let signal_number = libc::SIGRTMIN() + self.sig_rt_min_offset as libc::c_int;
         let mut sent_signal = false;
         let mut target_generation: Option<u64> = None;
 
         loop {
             let (running, generation) = self.get_running_and_generation();
 
+            // Stamp generation into cancel_requested if requested and this is the first iteration
+            // We stamp even when running=false to support killing during host function calls
+            // The generation tracking will prevent stale kills from affecting new calls
+            // Only stamp if a call is actually active (call_active=true)
+            if stamp_generation
+                && target_generation.is_none()
+                && self.call_active.load(Ordering::Acquire)
+            {
+                self.set_cancel_requested(generation);
+                target_generation = Some(generation);
+            }
+
+            // If not running, we've stamped the generation (if requested), so we're done
```

Comment:
should there be a check that `call_active` is `false` so this exits early in the case that the call ins't active? 

I think this is currently working due to the fact that when `call_active` `running` is also active. I think there is one case where the call might not be active but the vm is running (during initial set up) and i think it might hang in that scenario?